code-challenges.test.js

```diff
@@ -24,15 +24,29 @@ var secretCodeWord2 = "Gobbledygook"
 var secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe("codeWord", () => {
 
+    // the test method describes in plain words what the function should be doing.
+    test("returns the word if the word contains the letter a", () => {
+  
+      //the expect checks the expected output of the function return is what is actually returned. 
+      expect(codeWord(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(codeWord(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(codeWord(secretCodeWord3)).toEqual("3cc3ntr1c")
 
-
+    })
+  })
 
 // b) Create the function that makes the test pass.
+// USe the .replace method to include the characters to replace and what to change them to. 
+const codeWord = (string) => {
+  return string.replace(/[a,e,i,o]/g, 4, 3, 1, 0);
+}
+// // I'm able to get all the vowels replaced, but am having trouble passing through the correct replacement number. I'm only getting the 4 passed into the variables or the 0 but not the 3 or 1. 
```

Comment:
I'm fairly sure this has to do with ".replace" logic and how it interacts with strings and wants you to pass substring commands. I'd have to do a bit more research to fully understand it. Regex is great when it works, you could always just code it out with more basic logic in the mean time. Regex is great for refactoring